Userland/Applications/PixelPaint/MainWidget.cpp

```diff
@@ -1175,4 +1180,18 @@ void MainWidget::drop_event(GUI::DropEvent& event)
         open_image(response.value());
     }
 }
+
+void MainWidget::update_status_bar(DeprecatedString appended_text)
+{
+    if (!appended_text.is_empty()) {
+        StringBuilder builder = StringBuilder();
+        builder.append(m_last_image_editor_mouse_position.to_deprecated_string());
+        builder.append(" "sv);
+        builder.append(appended_text);
+        m_statusbar->set_override_text(builder.to_deprecated_string());
+    } else {
+        m_statusbar->set_override_text(m_last_image_editor_mouse_position.to_deprecated_string());
+    }
```

Comment:
What I mean is that you should create a StringBuilder outside the if-else, append the mouse position to it, then only append the appended_text if it's nonempty, then set the override_text to the string builder result in either case. Therefore, the if is just one small subcase of the general case which shares most of the code.

Reply:
Corrected, I understand what you mean now